tests/components/vera/common.py

```diff
@@ -0,0 +1,839 @@
+    assert attribute in state.attributes, f"Attribute '{attribute}' was not found."
+
+    value = state.attributes.get(attribute)
+    assert (
+        value == expected
+    ), f"Expected {attribute} '{expected}', but was '{value}' for entity '{state.entity_id}'"
+
+
+async def async_call_service(
+    hass: HomeAssistant,
+    data: ComponentData,
+    device_id: int,
+    platform: str,
+    service: str,
+    extra_attribs: dict = None,
+) -> None:
+    """Call an HA service for a vera device."""
+    await hass.services.async_call(
+        platform,
+        service,
+        {
+            **{"entity_id": get_entity_id(get_device(device_id, data), platform)},
+            **(extra_attribs or {}),
+        },
+    )
+
+    await hass.async_block_till_done()
+
+
+class ResponseStub:
+    """Simple stub."""
+
+    def __init__(self, json=None, text=None):
+        """Init object."""
+        self._json = json
+        self._text = text
+
+    def json(self) -> Any:
+        """Get json."""
+        return self._json
+
+    def text(self) -> str:
+        """Get text."""
+        return self._text
+
+
+async def async_configure_component(
+    hass: HomeAssistant,
+    response_sdata: dict,
+    response_status: dict,
+    respone_lu_sdata: dict,
+) -> ComponentData:
+    """Configure the component with specific mock data."""
+    controller_url = "http://127.0.0.1:123"
+
+    def data_request(payload: dict, timeout=TIMEOUT):
+        nonlocal component_data
+        payload_id = payload.get("id")
+
+        if payload_id == "sdata":
+            return ResponseStub(json=component_data.sdata)
+        if payload_id == "status":
+            return ResponseStub(json=component_data.status)
+        if payload_id == "lu_sdata":
+            return ResponseStub(json=component_data.lu_sdata)
+        if payload_id == "action":
+            return ResponseStub(json={})
+        if payload_id == "variableget":
+            device_id = int(payload.get("DeviceNum"))
+            variable = payload.get("Variable")
+
+            status = get_device_status(device_id, component_data)
+            for state in status.get("states", []):
+                if state.get("variable") == variable:
+                    return state.get("value")
+
+            return ResponseStub(text="")
+        if payload_id == "lu_action":
+            params = payload.copy()
+            params.pop("id")
+            service_id = params.pop("serviceId")
+            action = params.pop("action")
+            device_id = int(params.pop("DeviceNum"))
+            params.pop("output_format")
+            set_state_variable_name = next(
+                key for key in params if key.lower().startswith("new")
+            )
+            state_variable_name = set_state_variable_name[3:]
+            state_variable_value = params.pop(set_state_variable_name)
+            status_variable_name = None
+
+            if service_id == "urn:upnp-org:serviceId:SwitchPower1":
```

Comment:
The service_id is a detail of pyvera. We shouldn't need to interact with that. Please mock and spec the controller and devices instead.

Reply:
I think the tests should simulate the vera service as much as possible to ensure the greatest accuracy of end to end tests. The need here arises because no Dev will have real hardware to test against.

Reply:
Martin has a point.

- Tests that verify service data correctly maps to `pyvera.VeraSwitch` should be part of the pyvera test suite.
- Tests that verify `pyvera.VeraSwitch` <> `homeassistant.SwitchDevice` should be in Home Assistant.

By splitting the tests at those points, both test suites will also be easier to follow.